feat: implement RFC 16 to allow emergency node access (#3557)
edgelesssys/constellation: docs/docs/workflows/troubleshooting.md

Reviewer: I'd suggest @thomasten or @m1ghtym0 skim over these logs quickly to read from an "external" perspective and make sure this is understandable.

1. Figure out which node to connect to:

   ```bash
   kubectl get nodes
   # or to see more information, such as IPs:
   kubectl get nodes -o wide
   ```

2. Connect to the node:

   ```bash
   kubectl debug node/constell-worker-xksa0-000000 -it --image=busybox
   ```

   You will be presented with a prompt.

>    The nodes file system is mounted at `/host`.

Reviewer: ```suggestion
   The file system of the node is mounted at `/host`.
```

Reviewer: Since that line is touched here anyways, I'd just correct it. But you can also leave it as-is. Again, no strong opinion :p

3. Once finished, clean up the debug pod:

   ```bash
   kubectl delete pod node-debugger-constell-worker-xksa0-000000-bjthj
   ```

### Emergency SSH access

Emergency SSH access to nodes can be useful to diagnose issues or download important data even if the Kubernetes API isn't reachable anymore.

> 1. Enter the `constellation-terraform` directory in your constellation workspace and enable emergency SSH access to the cluster:

Reviewer: ```suggestion
1. Enter the `constellation-terraform` directory in your Constellation workspace and enable emergency SSH access to the cluster:
```
nit

   ```bash
    cd constellation-terraform
    echo "emergency_ssh = true" >> ./terraform.tfvars
    terraform apply
   ```

2. Sign an existing SSH key with your master secret:

   ```bash
   cd ../ # go back to your constellation workspace
   constellation ssh --key your_public_key.pub
   ```

   A certificate is written to `constellation_cert.pub`.

   The certificate is valid for 24 hours and enables you to access your constellation nodes using
   [certificate based authentication](https://en.wikibooks.org/wiki/OpenSSH/Cookbook/Certificate-based_Authentication).

3. Finally, you can connect to any constellation node using your certificate and your private key.

   `ssh -o CertificateFile=constellation_cert.pub -i <your private key> root@<ip of constellation node>`

   Normally, you won't have access to all constellation nodes since they reside in a private network.

>    To access those nodes anyways, you can use your constellation load balancer as a proxy jump host.

Reviewer: ```suggestion
   To access those nodes anyways, you can use your Constellation load balancer as a proxy jump host.
```
nit

   For this, use something along the following ssh client configuration:

   ```text
   Host <LB domain name>
     ProxyJump none

   Host *
     IdentityFile <your private key>
     PreferredAuthentications publickey
     CertificateFile=constellation_cert.pub
     User root
     ProxyJump <LB domain name>
   ```

   Using this config you can connect to a constellation node using `ssh -F <this config> <private node IP>`.

>    You can obtain the private node IP and the domain name of the load balancer using your CSPs web UI.

Reviewer: ```suggestion
   You can obtain the private node IP and the domain name of the load balancer using your CSP's web UI.
```

    ```bash
    kubectl delete pod node-debugger-constell-worker-xksa0-000000-bjthj
    ```

### Emergency SSH access

Emergency SSH access to nodes can be useful to diagnose issues or download important data even if the Kubernetes API isn't reachable anymore.

1. Enter the `constellation-terraform` directory in your Constellation workspace and enable emergency SSH access to the cluster:

   ```bash
    cd constellation-terraform
    echo "emergency_ssh = true" >> ./terraform.tfvars
    terraform apply
   ```

2. Sign an existing SSH key with your master secret:

   ```bash

>    cd ../ # go back to your constellation workspace
>    constellation ssh --key your_public_key.pub
>    ```
> 
>    A certificate is written to `constellation_cert.pub`.
> 
>    The certificate is valid for 24 hours and enables you to access your constellation nodes using

Reviewer 2: ````suggestion
   cd ../ # go back to your Constellation workspace
   constellation ssh --key your_public_key.pub
   ```

   A certificate is written to `constellation_cert.pub`.

   The certificate is valid for 24 hours and enables you to access your Constellation nodes using
````

   [certificate based authentication](https://en.wikibooks.org/wiki/OpenSSH/Cookbook/Certificate-based_Authentication).

> 3. Finally, you can connect to any constellation node using your certificate and your private key.

Reviewer 2: ```suggestion
3. Now you can connect to any Constellation node using your certificate and private key.
```

>    `ssh -o CertificateFile=constellation_cert.pub -i <your private key> root@<ip of constellation node>`

Reviewer 2: ````suggestion
   ```bash
   ssh -o CertificateFile=constellation_cert.pub -i <your private key> root@<ip of constellation node>
   ```
````

>    Normally, you won't have access to all constellation nodes since they reside in a private network.
>    To access those nodes anyways, you can use your Constellation load balancer as a proxy jump host.
>    For this, use something along the following ssh client configuration:

Reviewer 2: ```suggestion
   Normally, you don't have access to the Constellation nodes since they reside in a private network.
   To access those nodes, you can use your Constellation load balancer as a proxy jump host.
   For this, use something along the following SSH client configuration:
```

   ```text
   Host <LB domain name>
     ProxyJump none

   Host *
     IdentityFile <your private key>
     PreferredAuthentications publickey
     CertificateFile=constellation_cert.pub
     User root
     ProxyJump <LB domain name>
   ```

>    Using this config you can connect to a constellation node using `ssh -F <this config> <private node IP>`.

Reviewer 2: ```suggestion
   With this configuration, you can connect to a Constellation node using `ssh -F <this config> <private node IP>`.
```
